### ISSUE TITLE
WIC Proxy Functions?

### ISSUE THREAD

**Issue by GCuser99** [fafalone/WinDevLib#22]
such as:
```vba
Private Declare Function IWICImagingFactory_CreateDecoderFromFilename_Proxy Lib "windowscodecs" (ByVal pFactory As stdole.IUnknown, ByVal wzFilename As Long, pguidVendor As Any, ByVal dwDesiredAccess As Long, ByVal lMetadataOptions As Long, ppIDecoder As stdole.IUnknown) As Long
```
List can be found here: https://learn.microsoft.com/en-us/windows/win32/wic/wic-proxy-functions

Thx!

**Comment by fafalone** [comment]
Seems a bit redundant with all the interfaces defined, but sure. For the next few days I want to first finish something else I started (imagehlp/dbghelp APIs), but after that I'll start on these.

**Comment by GCuser99** [comment]
No hurry nor a must include. I ran across the api in a [legacy VB6 project](https://github.com/wqweto/AlphaBlendImage/blob/master/src/AlphaBlendImage.ctl) while researching an image processing need of mine but found what I needed elsewhere. Just thought I should report it for completeness sake. Thanks.

**Comment by fafalone** [comment]
It's going to take me a lot longer than I thought to slog through what I was previously in the middle of, but it turns out it was really easy to add these because there's only a select few of them (every method of every interface isn't covered, quite a small percentage actually, so if you want to use them in advance of the next release, here they are:

```
Public Declare PtrSafe Function IEnumString_Next_WIC_Proxy Lib "Windowscodecs.dll" (ByVal this As stdole.IUnknown, ByVal celt As Long, rgelt As LongPtr, pceltFetched As Long) As Long
Public Declare PtrSafe Function IEnumString_Reset_WIC_Proxy Lib "Windowscodecs.dll" (ByVal this As stdole.IUnknown) As Long
Public Declare PtrSafe Function IPropertyBag2_Write_Proxy Lib "Windowscodecs.dll" (ByVal this As stdole.IUnknown, ByVal cProperties As Long, pPropBag As PROPBAG2, ByVal pvarValue As LongPtr) As Long
Public Declare PtrSafe Function IWICBitmapClipper_Initialize_Proxy Lib "Windowscodecs.dll" (ByVal this As stdole.IUnknown, ByVal pISource As stdole.IUnknown, prc As WICRect) As Long
Public Declare PtrSafe Function IWICBitmapCodecInfo_DoesSupportAnimation_Proxy Lib "Windowscodecs.dll" (ByVal this As stdole.IUnknown, ByRef pfSupportAnimation As BOOL) As Long
Public Declare PtrSafe Function IWICBitmapCodecInfo_DoesSupportLossless_Proxy Lib "Windowscodecs.dll" (ByVal this As stdole.IUnknown, ByRef pfSupportLossless As BOOL) As Long
Public Declare PtrSafe Function IWICBitmapCodecInfo_DoesSupportMultiframe_Proxy Lib "Windowscodecs.dll" (ByVal this As stdole.IUnknown, ByRef pfSupportMultiframe As BOOL) As Long
Public Declare PtrSafe Function IWICBitmapCodecInfo_GetContainerFormat_Proxy Lib "Windowscodecs.dll" (ByVal this As stdole.IUnknown, ByRef pguidContainerFormat As UUID) As Long
Public Declare PtrSafe Function IWICBitmapCodecInfo_GetDeviceManufacturer_Proxy Lib "Windowscodecs.dll" (ByVal this As stdole.IUnknown, ByVal cchDeviceManufacturer As Long, ByVal wzDeviceManufacturer As LongPtr, ByRef pcchActual As Long) As Long
Public Declare PtrSafe Function IWICBitmapCodecInfo_GetDeviceModels_Proxy Lib "Windowscodecs.dll" (ByVal this As stdole.IUnknown, ByVal cchDeviceModels As Long, ByVal wzDeviceModels As LongPtr, ByRef pcchActual As Long) As Long
Public Declare PtrSafe Function IWICBitmapCodecInfo_GetFileExtensions_Proxy Lib "Windowscodecs.dll" (ByVal this As stdole.IUnknown, ByVal cchFileExtensions As Long, ByVal wzFileExtensions As LongPtr, ByRef pcchActual As Long) As Long
Public Declare PtrSafe Function IWICBitmapCodecInfo_GetMimeTypes_Proxy Lib "Windowscodecs.dll" (ByVal this As stdole.IUnknown, ByVal cchMimeTypes As Long, ByVal wzMimeTypes As LongPtr, ByRef pcchActual As Long) As Long
Public Declare PtrSafe Function IWICBitmapDecoder_CopyPalette_Proxy Lib "Windowscodecs.dll" (ByVal this As stdole.IUnknown, ByVal pIPalette As stdole.IUnknown) As Long
Public Declare PtrSafe Function IWICBitmapDecoder_GetColorContexts_Proxy Lib "Windowscodecs.dll" (ByVal this As stdole.IUnknown, ByVal cCount As Long, ByRef ppIColorContexts As stdole.IUnknown, ByRef pcActualCount As Long) As Long
Public Declare PtrSafe Function IWICBitmapDecoder_GetDecoderInfo_Proxy Lib "Windowscodecs.dll" (ByVal this As stdole.IUnknown, ByRef ppIDecoderInfo As stdole.IUnknown) As Long
Public Declare PtrSafe Function IWICBitmapDecoder_GetFrameCount_Proxy Lib "Windowscodecs.dll" (ByVal this As stdole.IUnknown, ByRef pCount As Long) As Long
Public Declare PtrSafe Function IWICBitmapDecoder_GetFrame_Proxy Lib "Windowscodecs.dll" (ByVal this As stdole.IUnknown, ByVal index As Long, ppIBitmapFrame As stdole.IUnknown) As Long
Public Declare PtrSafe Function IWICBitmapDecoder_GetMetadataQueryReader_Proxy Lib "Windowscodecs.dll" (ByVal this As stdole.IUnknown, ByRef ppIMetadataQueryReader As stdole.IUnknown) As Long
Public Declare PtrSafe Function IWICBitmapDecoder_GetPreview_Proxy Lib "Windowscodecs.dll" (ByVal this As stdole.IUnknown, ByRef ppIBitmapSource As stdole.IUnknown) As Long
Public Declare PtrSafe Function IWICBitmapDecoder_GetThumbnail_Proxy Lib "Windowscodecs.dll" (ByVal this As stdole.IUnknown, ByRef ppIThumbnail As stdole.IUnknown) As Long
Public Declare PtrSafe Function IWICBitmapEncoder_Commit_Proxy Lib "Windowscodecs.dll" (ByVal this As stdole.IUnknown) As Long
Public Declare PtrSafe Function IWICBitmapEncoder_CreateNewFrame_Proxy Lib "Windowscodecs.dll" (ByVal this As stdole.IUnknown, ByRef ppIFrameEncode As stdole.IUnknown, ByRef ppIEncoderOptions As stdole.IUnknown) As Long
Public Declare PtrSafe Function IWICBitmapEncoder_GetEncoderInfo_Proxy Lib "Windowscodecs.dll" (ByVal this As stdole.IUnknown, ByRef ppIEncoderInfo As stdole.IUnknown) As Long
Public Declare PtrSafe Function IWICBitmapEncoder_GetMetadataQueryWriter_Proxy Lib "Windowscodecs.dll" (ByVal this As stdole.IUnknown, ByRef ppIMetadataQueryWriter As stdole.IUnknown) As Long
Public Declare PtrSafe Function IWICBitmapEncoder_Initialize_Proxy Lib "Windowscodecs.dll" (ByVal this As stdole.IUnknown, ByVal pIStream As stdole.IUnknown, ByVal cacheOption As WICBitmapEncoderCacheOption) As Long
Public Declare PtrSafe Function IWICBitmapEncoder_SetPalette_Proxy Lib "Windowscodecs.dll" (ByVal this As stdole.IUnknown, ByVal pIPalette As stdole.IUnknown) As Long
Public Declare PtrSafe Function IWICBitmapEncoder_SetThumbnail_Proxy Lib "Windowscodecs.dll" (ByVal this As stdole.IUnknown, ByVal pIThumbnail As stdole.IUnknown) As Long
Public Declare PtrSafe Function IWICBitmapFlipRotator_Initialize_Proxy Lib "Windowscodecs.dll" (ByVal this As stdole.IUnknown, ByVal pISource As stdole.IUnknown, ByVal Options As WICBitmapTransformOptions) As Long
Public Declare PtrSafe Function IWICBitmapFrameDecode_GetColorContexts_Proxy Lib "Windowscodecs.dll" (ByVal this As stdole.IUnknown, ByVal cCount As Long, ByRef ppIColorContexts As stdole.IUnknown, ByRef pcActualCount As Long) As Long
Public Declare PtrSafe Function IWICBitmapFrameDecode_GetMetadataQueryReader_Proxy Lib "Windowscodecs.dll" (ByVal this As stdole.IUnknown, ByRef ppIMetadataQueryReader As stdole.IUnknown) As Long
Public Declare PtrSafe Function IWICBitmapFrameDecode_GetThumbnail_Proxy Lib "Windowscodecs.dll" (ByVal this As stdole.IUnknown, ByRef ppIThumbnail As stdole.IUnknown) As Long
Public Declare PtrSafe Function IWICBitmapFrameEncode_Commit_Proxy Lib "Windowscodecs.dll" (ByVal this As stdole.IUnknown) As Long
Public Declare PtrSafe Function IWICBitmapFrameEncode_GetMetadataQueryWriter_Proxy Lib "Windowscodecs.dll" (ByVal this As stdole.IUnknown, ByRef ppIMetadataQueryWriter As stdole.IUnknown) As Long
Public Declare PtrSafe Function IWICBitmapFrameEncode_Initialize_Proxy Lib "Windowscodecs.dll" (ByVal this As stdole.IUnknown, ByVal pIEncoderOptions As stdole.IUnknown) As Long
Public Declare PtrSafe Function IWICBitmapFrameEncode_SetColorContexts_Proxy Lib "Windowscodecs.dll" (ByVal this As stdole.IUnknown, ByVal cCount As Long, ByRef ppIColorContext As stdole.IUnknown) As Long
Public Declare PtrSafe Function IWICBitmapFrameEncode_SetResolution_Proxy Lib "Windowscodecs.dll" (ByVal this As stdole.IUnknown, ByVal DpiX As Double, ByVal DpiY As Double) As Long
Public Declare PtrSafe Function IWICBitmapFrameEncode_SetSize_Proxy Lib "Windowscodecs.dll" (ByVal this As stdole.IUnknown, ByVal uiWidth As Long, ByVal uiHeight As Long) As Long
Public Declare PtrSafe Function IWICBitmapFrameEncode_SetThumbnail_Proxy Lib "Windowscodecs.dll" (ByVal this As stdole.IUnknown, ByVal pIThumbnail As stdole.IUnknown) As Long
Public Declare PtrSafe Function IWICBitmapFrameEncode_WriteSource_Proxy Lib "Windowscodecs.dll" (ByVal this As stdole.IUnknown, ByVal pIBitmapSource As stdole.IUnknown, prc As WICRect) As Long
Public Declare PtrSafe Function IWICBitmapLock_GetStride_Proxy Lib "Windowscodecs.dll" (ByVal this As stdole.IUnknown, ByRef pcbStride As Long) As Long
Public Declare PtrSafe Function IWICBitmapLock_GetDataPointer_STA_Proxy Lib "Windowscodecs.dll" (ByVal this As stdole.IUnknown, ByRef pcbBufferSize As Long, ByRef ppbData As LongPtr) As Long
Public Declare PtrSafe Function IWICBitmapScaler_Initialize_Proxy Lib "Windowscodecs.dll" (ByVal this As stdole.IUnknown, ByVal pISource As stdole.IUnknown, ByVal uiWidth As Long, ByVal uiHeight As Long, ByVal mode As WICBitmapInterpolationMode) As Long
Public Declare PtrSafe Function IWICBitmapSource_GetSize_Proxy Lib "Windowscodecs.dll" (ByVal this As stdole.IUnknown, ByRef puiWidth As Long, ByRef puiHeight As Long) As Long
Public Declare PtrSafe Function IWICBitmapSource_GetPixelFormat_Proxy Lib "Windowscodecs.dll" (ByVal this As stdole.IUnknown, ByRef pPixelFormat As UUID) As Long
Public Declare PtrSafe Function IWICBitmapSource_GetResolution_Proxy Lib "Windowscodecs.dll" (ByVal this As stdole.IUnknown, ByRef pDpiX As Double, ByRef pDpiY As Double) As Long
Public Declare PtrSafe Function IWICBitmapSource_CopyPalette_Proxy Lib "Windowscodecs.dll" (ByVal this As stdole.IUnknown, ByVal pIPalette As stdole.IUnknown) As Long
Public Declare PtrSafe Function IWICBitmapSource_CopyPixels_Proxy Lib "Windowscodecs.dll" (ByVal this As stdole.IUnknown, prc As WICRect, ByVal cbStride As Long, ByVal cbBufferSize As Long, pbBuffer As Any) As Long
Public Declare PtrSafe Function IWICBitmap_Lock_Proxy Lib "Windowscodecs.dll" (ByVal this As stdole.IUnknown, prcLock As WICRect, ByVal flags As WICBitmapLockFlags, ppILock As stdole.IUnknown) As Long
Public Declare PtrSafe Function IWICBitmap_SetPalette_Proxy Lib "Windowscodecs.dll" (ByVal this As stdole.IUnknown, ByVal pIPalette As stdole.IUnknown) As Long
Public Declare PtrSafe Function IWICBitmap_SetResolution_Proxy Lib "Windowscodecs.dll" (ByVal this As stdole.IUnknown, ByVal DpiX As Double, ByVal DpiY As Double) As Long
Public Declare PtrSafe Function IWICColorContext_InitializeFromMemory_Proxy Lib "Windowscodecs.dll" (ByVal this As stdole.IUnknown, pbBuffer As Any, ByVal cbNormalSize As Long) As Long
Public Declare PtrSafe Function IWICComponentFactory_CreateMetadataWriterFromReader_Proxy Lib "Windowscodecs.dll" (ByVal this As stdole.IUnknown, ByVal pIReader As stdole.IUnknown, ByRef pguidVendor As UUID, ByRef ppIWriter As stdole.IUnknown) As Long
Public Declare PtrSafe Function IWICComponentFactory_CreateQueryWriterFromBlockWriter_Proxy Lib "Windowscodecs.dll" (ByVal this As stdole.IUnknown, ByVal pIBlockWriter As stdole.IUnknown, ByRef ppIQueryWriter As stdole.IUnknown) As Long
Public Declare PtrSafe Function IWICComponentInfo_GetAuthor_Proxy Lib "Windowscodecs.dll" (ByVal this As stdole.IUnknown, ByVal cchAuthor As Long, ByVal wzAuthhor As LongPtr, ByRef pcchActual As Long) As Long
Public Declare PtrSafe Function IWICComponentInfo_GetCLSID_Proxy Lib "Windowscodecs.dll" (ByVal this As stdole.IUnknown, ByRef pclsid As UUID) As Long
Public Declare PtrSafe Function IWICComponentInfo_GetFriendlyName_Proxy Lib "Windowscodecs.dll" (ByVal this As stdole.IUnknown, ByVal cchFriendlyName As Long, ByVal wzFriendlyName As LongPtr, ByRef pcchActual As Long) As Long
Public Declare PtrSafe Function IWICComponentInfo_GetSpecVersion_Proxy Lib "Windowscodecs.dll" (ByVal this As stdole.IUnknown, ByVal cchSpecVersion As Long, ByVal wzSpecVersion As LongPtr, ByRef pcchActual As Long) As Long
Public Declare PtrSafe Function IWICComponentInfo_GetVersion_Proxy Lib "Windowscodecs.dll" (ByVal this As stdole.IUnknown, ByVal cchVersion As Long, ByVal wzVersion As LongPtr, ByRef pcchActual As Long) As Long
Public Declare PtrSafe Function IWICFastMetadataEncoder_Commit_Proxy Lib "Windowscodecs.dll" (ByVal this As stdole.IUnknown) As Long
Public Declare PtrSafe Function IWICFastMetadataEncoder_GetMetadataQueryWriter_Proxy Lib "Windowscodecs.dll" (ByVal this As stdole.IUnknown, ByRef ppIMetadataQueryWriter As stdole.IUnknown) As Long
Public Declare PtrSafe Function IWICFormatConverter_Initialize_Proxy Lib "Windowscodecs.dll" (ByVal this As stdole.IUnknown, ByVal pISource As stdole.IUnknown, ByRef dstFormat As UUID, ByVal dither As WICBitmapDitherType, ByVal pIPalette As stdole.IUnknown, ByVal alphaThresholdPercent As Double, ByVal paletteTranslate As WICBitmapPaletteType) As Long
Public Declare PtrSafe Function IWICImagingFactory_CreateBitmapClipper_Proxy Lib "Windowscodecs.dll" (ByVal this As stdole.IUnknown, ByRef ppIBitmapClipper As stdole.IUnknown) As Long
Public Declare PtrSafe Function IWICImagingFactory_CreateBitmapFlipRotator_Proxy Lib "Windowscodecs.dll" (ByVal this As stdole.IUnknown, ByRef ppIBitmapFlipRotator As stdole.IUnknown) As Long
Public Declare PtrSafe Function IWICImagingFactory_CreateBitmapFromHBITMAP_Proxy Lib "Windowscodecs.dll" (ByVal this As stdole.IUnknown, ByVal hbitmap As LongPtr, ByVal hPalette As LongPtr, ByVal Options As WICBitmapAlphaChannelOption, ByRef ppIBitmap As stdole.IUnknown) As Long
Public Declare PtrSafe Function IWICImagingFactory_CreateBitmapFromHICON_Proxy Lib "Windowscodecs.dll" (ByVal this As stdole.IUnknown, ByVal hIcon As LongPtr, ByRef ppIBitmap As stdole.IUnknown) As Long
Public Declare PtrSafe Function IWICImagingFactory_CreateBitmapFromMemory_Proxy Lib "Windowscodecs.dll" (ByVal this As stdole.IUnknown, ByVal uiWidth As Long, ByVal uiHeight As Long, ByRef PixelFormat As UUID, ByVal cbStride As Long, ByVal cbBufferSize As Long, pbBuffer As Any, ByRef ppIBitmap As stdole.IUnknown) As Long
Public Declare PtrSafe Function IWICImagingFactory_CreateBitmapFromSource_Proxy Lib "Windowscodecs.dll" (ByVal this As stdole.IUnknown, ByVal pIBitmapSource As stdole.IUnknown, ByVal option As WICBitmapCreateCacheOption, ByRef ppIBitmap As stdole.IUnknown) As Long
Public Declare PtrSafe Function IWICImagingFactory_CreateBitmapScaler_Proxy Lib "Windowscodecs.dll" (ByVal this As stdole.IUnknown, ByRef ppIBitmapScaler As stdole.IUnknown) As Long
Public Declare PtrSafe Function IWICImagingFactory_CreateBitmap_Proxy Lib "Windowscodecs.dll" (ByVal this As stdole.IUnknown, ByVal uiWidth As Long, ByVal uiHeight As Long, ByRef PixelFormat As UUID, ByVal option As WICBitmapCreateCacheOption, ByRef ppIBitmap As stdole.IUnknown) As Long
Public Declare PtrSafe Function IWICImagingFactory_CreateComponentInfo_Proxy Lib "Windowscodecs.dll" (ByVal this As stdole.IUnknown, ByRef clsidComponent As UUID, ByRef ppIInfo As stdole.IUnknown) As Long
Public Declare PtrSafe Function IWICImagingFactory_CreateDecoderFromFileHandle_Proxy Lib "Windowscodecs.dll" (ByVal this As stdole.IUnknown, ByVal hFile As LongPtr, ByRef pguidVendor As UUID, ByVal metadataOptions As WICDecodeOptions, ppIDecoder As stdole.IUnknown) As Long
Public Declare PtrSafe Function IWICImagingFactory_CreateDecoderFromFilename_Proxy Lib "Windowscodecs.dll" (ByVal this As stdole.IUnknown, ByVal wzFilename As LongPtr, ByRef pguidVendor As UUID, ByVal dwDesiredAccess As Long, ByVal metadataOptions As WICDecodeOptions, ppIDecoder As stdole.IUnknown) As Long
Public Declare PtrSafe Function IWICImagingFactory_CreateDecoderFromStream_Proxy Lib "Windowscodecs.dll" (ByVal this As stdole.IUnknown, ByVal pIStream As stdole.IUnknown, ByRef pguidVendor As UUID, ByVal metadataOptions As WICDecodeOptions, ppIDecoder As stdole.IUnknown) As Long
Public Declare PtrSafe Function IWICImagingFactory_CreateEncoder_Proxy Lib "Windowscodecs.dll" (ByVal this As stdole.IUnknown, ByRef guidContainerFormat As UUID, ByRef pguidVendor As UUID, ppIEncoder As stdole.IUnknown) As Long
Public Declare PtrSafe Function IWICImagingFactory_CreateFastMetadataEncoderFromDecoder_Proxy Lib "Windowscodecs.dll" (ByVal this As stdole.IUnknown, ByVal pIDecoder As stdole.IUnknown, ByRef ppIFastEncoder As stdole.IUnknown) As Long
Public Declare PtrSafe Function IWICImagingFactory_CreateFastMetadataEncoderFromFrameDecode_Proxy Lib "Windowscodecs.dll" (ByVal this As stdole.IUnknown, ByVal pIFrameDecoder As stdole.IUnknown, ByRef ppIFastEncoder As stdole.IUnknown) As Long
Public Declare PtrSafe Function IWICImagingFactory_CreateFormatConverter_Proxy Lib "Windowscodecs.dll" (ByVal this As stdole.IUnknown, ppIFormatConverter As stdole.IUnknown) As Long
Public Declare PtrSafe Function IWICImagingFactory_CreatePalette_Proxy Lib "Windowscodecs.dll" (ByVal this As stdole.IUnknown, ByRef ppIPalette As stdole.IUnknown) As Long
Public Declare PtrSafe Function IWICImagingFactory_CreateQueryWriterFromReader_Proxy Lib "Windowscodecs.dll" (ByVal this As stdole.IUnknown, ByVal pIQueryReader As stdole.IUnknown, ByRef pguidVendor As UUID, ByRef ppIQueryWriter As stdole.IUnknown) As Long
Public Declare PtrSafe Function IWICImagingFactory_CreateQueryWriter_Proxy Lib "Windowscodecs.dll" (ByVal this As stdole.IUnknown, ByRef guidMetadataFormat As UUID, ByRef pguidVendor As UUID, ByRef ppIQueryWriter As stdole.IUnknown) As Long
Public Declare PtrSafe Function IWICImagingFactory_CreateStream_Proxy Lib "Windowscodecs.dll" (ByVal this As stdole.IUnknown, ByRef ppIWICStream As stdole.IUnknown) As Long
Public Declare PtrSafe Function IWICMetadataBlockReader_GetCount_Proxy Lib "Windowscodecs.dll" (ByVal this As stdole.IUnknown, ByRef pcCount As Long) As Long
Public Declare PtrSafe Function IWICMetadataBlockReader_GetReaderByIndex_Proxy Lib "Windowscodecs.dll" (ByVal this As stdole.IUnknown, ByVal nIndex As Long, ByRef ppIMetadataReader As stdole.IUnknown) As Long
Public Declare PtrSafe Function IWICMetadataQueryReader_GetContainerFormat_Proxy Lib "Windowscodecs.dll" (ByVal this As stdole.IUnknown, ByRef pguidContainerFormat As UUID) As Long
Public Declare PtrSafe Function IWICMetadataQueryReader_GetEnumerator_Proxy Lib "Windowscodecs.dll" (ByVal this As stdole.IUnknown, ByRef ppIEnumString As stdole.IUnknown) As Long
Public Declare PtrSafe Function IWICMetadataQueryReader_GetLocation_Proxy Lib "Windowscodecs.dll" (ByVal this As stdole.IUnknown, ByVal cchMaxLength As Long, ByVal wzNamespace As LongPtr, ByRef pcchActualLength As Long) As Long
Public Declare PtrSafe Function IWICMetadataQueryReader_GetMetadataByName_Proxy Lib "Windowscodecs.dll" (ByVal this As stdole.IUnknown, ByVal wzName As LongPtr, ByRef pvarValue As Variant) As Long
Public Declare PtrSafe Function IWICMetadataQueryWriter_RemoveMetadataByName_Proxy Lib "Windowscodecs.dll" (ByVal this As stdole.IUnknown, ByVal wzName As LongPtr) As Long
Public Declare PtrSafe Function IWICMetadataQueryWriter_SetMetadataByName_Proxy Lib "Windowscodecs.dll" (ByVal this As stdole.IUnknown, ByVal wzName As LongPtr, ByRef pvarValue As Variant) As Long
Public Declare PtrSafe Function IWICPalette_GetColorCount_Proxy Lib "Windowscodecs.dll" (ByVal this As stdole.IUnknown, ByRef pcCount As Long) As Long
Public Declare PtrSafe Function IWICPalette_GetColors_Proxy Lib "Windowscodecs.dll" (ByVal this As stdole.IUnknown, ByVal cCount As Long, ByRef pColors As Long, ByRef pcActualColors As Long) As Long
Public Declare PtrSafe Function IWICPalette_GetType_Proxy Lib "Windowscodecs.dll" (ByVal this As stdole.IUnknown, ByRef pePaletteType As WICBitmapPaletteType) As Long
Public Declare PtrSafe Function IWICPalette_HasAlpha_Proxy Lib "Windowscodecs.dll" (ByVal this As stdole.IUnknown, ByRef pfHasAlpha As BOOL) As Long
Public Declare PtrSafe Function IWICPalette_InitializeCustom_Proxy Lib "Windowscodecs.dll" (ByVal this As stdole.IUnknown, ByRef pColors As Long, ByVal cCount As Long) As Long
Public Declare PtrSafe Function IWICPalette_InitializeFromBitmap_Proxy Lib "Windowscodecs.dll" (ByVal this As stdole.IUnknown, ByVal pISurface As stdole.IUnknown, ByVal cCount As Long, ByVal fAddTransparentColor As BOOL) As Long
Public Declare PtrSafe Function IWICPalette_InitializeFromPalette_Proxy Lib "Windowscodecs.dll" (ByVal this As stdole.IUnknown, ByVal pIPalette As stdole.IUnknown) As Long
Public Declare PtrSafe Function IWICPalette_InitializePredefined_Proxy Lib "Windowscodecs.dll" (ByVal this As stdole.IUnknown, ByVal ePaletteType As WICBitmapPaletteType, ByVal fAddTransparentColor As BOOL) As Long
Public Declare PtrSafe Function IWICPixelFormatInfo_GetBitsPerPixel_Proxy Lib "Windowscodecs.dll" (ByVal this As stdole.IUnknown, ByRef puiBitsPerPixel As Long) As Long
Public Declare PtrSafe Function IWICPixelFormatInfo_GetChannelCount_Proxy Lib "Windowscodecs.dll" (ByVal this As stdole.IUnknown, ByRef puiChannelCount As Long) As Long
Public Declare PtrSafe Function IWICPixelFormatInfo_GetChannelMask_Proxy Lib "Windowscodecs.dll" (ByVal this As stdole.IUnknown, ByVal uiChannelIndex As Long, ByVal cbMaskBuffer As Long, pbMaskBuffer As Any, ByRef pcbActual As Long) As Long
Public Declare PtrSafe Function IWICStream_InitializeFromIStream_Proxy Lib "Windowscodecs.dll" (ByVal this As stdole.IUnknown, ByVal pIStream As stdole.IUnknown) As Long
Public Declare PtrSafe Function IWICStream_InitializeFromMemory_Proxy Lib "Windowscodecs.dll" (ByVal this As stdole.IUnknown, ByRef pbBuffer As Any, ByVal cbBufferSize As Long) As Long
```

**Comment by GCuser99** [comment]
thanks Jon!

**Comment by fafalone** [comment]
These APIs are now available as of v7.10.396.

Thanks for requesting it; being in use by an existing VB6 project (or one you're planning on making) is indeed a very strong case for including them.  👍
